src/Acuminator/Acuminator.Analyzers/StaticAnalysis/EventHandlerModifier/EventHandlerModifierAnalyzer.cs

```diff
@@ -0,0 +1,140 @@
+using Acuminator.Utilities.Roslyn.Semantic.PXGraph;
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Collections.Generic;
+
+namespace Acuminator.Analyzers.StaticAnalysis.PrivateEventHandlers
+{
+	public class EventHandlerModifierAnalyzer : PXGraphAggregatedAnalyzerBase
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(
+				Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual,
+				Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations);
+
+		public override void Analyze(SymbolAnalysisContext context, PXContext pxContext, PXGraphEventSemanticModel graphExtension)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			foreach (var handler in graphExtension.GetAllEvents().Where(graphEvent => graphEvent.Symbol.IsDeclaredInType(graphExtension.Symbol)))
+			{
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				var location = handler.Symbol.Locations.FirstOrDefault();
+
+				if (location == null)
+				{
+					continue;
+				}
+
+				if (handler.Symbol.IsOverride)
+				{
+					continue;
+				}
+
+				var properties = new Dictionary<string, string?>
+				{
+					{ "IsOverride",							handler.Symbol.IsOverride.ToString()},
+					{ "IsExplicitInterfaceImplementation", (handler.Symbol.MethodKind == MethodKind.ExplicitInterfaceImplementation).ToString()},
+					{ "ImplementsInterface",				HelperMethods.ImplementsInterface(handler.Symbol, pxContext).ToString()},
+					{ "IsContainingTypeSealed",				handler.Symbol.ContainingType.IsSealed.ToString()}
+				}
+				.ToImmutableDictionary();
+
+				if (handler.Symbol.MethodKind == MethodKind.ExplicitInterfaceImplementation)
+				{
+					context.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations, location, properties));
+				}
+				else if (!HelperMethods.ImplementsInterface(handler.Symbol, pxContext))
+				{
+					var targetAccessibility = handler.Symbol.ContainingType.IsSealed ? Accessibility.Public : Accessibility.Protected;
+
+					if (!(handler.Symbol.IsVirtual || handler.Symbol.ContainingType.IsSealed) || handler.Symbol.DeclaredAccessibility != targetAccessibility)
+					{
+						context.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual, location, properties));
+					}
+				}
+			}
+		}
+
+		private static class HelperMethods
+		{
+			internal static bool ImplementsInterface(IMethodSymbol method, PXContext pxContext)
+			{
+				return method
+					.ContainingType
+					.AllInterfaces
+					.SelectMany(i => i.GetMethods(method.Name))
+					.Any(m =>
+						IsEventHandlerInfosMatch(method, m, pxContext) &&
+						IsImplicitInterfaceImplementation(method, m)
+					);
+			}
+
+			private static bool IsEventHandlerInfosMatch(IMethodSymbol method, IMethodSymbol other, PXContext pxContext)
```

Comment:
I feel that this is a confusing name. `IsValidEventHandlerSignature` sounds much clearer